src/ui/inspector/PlotElementsPool.vue

```diff
@@ -0,0 +1,333 @@
+            this.showSelection(selection);
+        }
+
+        this.openmct.selection.on('change', this.showSelection);
+        this.openmct.editor.on('isEditing', this.setEditState);
+    },
+    destroyed() {
+        this.openmct.editor.off('isEditing', this.setEditState);
+        this.openmct.selection.off('change', this.showSelection);
+
+        if (this.compositionUnlistener) {
+            this.compositionUnlistener();
+        }
+    },
+    methods: {
+        setEditState(isEditing) {
+            this.isEditing = isEditing;
+            this.showSelection(this.openmct.selection.get());
+        },
+        showSelection(selection) {
+            if (_.isEqual(this.selection, selection)) {
+                return;
+            }
+
+            this.selection = selection;
+            this.elementsCache = {};
+            this.listeners = [];
+            this.parentObject = selection && selection[0] && selection[0][0].context.item;
+
+            if (this.compositionUnlistener) {
+                this.compositionUnlistener();
+            }
+
+            if (this.parentObject) {
+                this.composition = this.openmct.composition.get(this.parentObject);
+
+                if (this.composition) {
+                    this.composition.load();
+
+                    this.composition.on('add', this.addElement);
+                    this.composition.on('remove', this.removeElement);
+                    this.composition.on('reorder', this.reorderElements);
+
+                    this.compositionUnlistener = () => {
+                        this.composition.off('add', this.addElement);
+                        this.composition.off('remove', this.removeElement);
+                        this.composition.off('reorder', this.reorderElements);
+                        delete this.compositionUnlistener;
+                    };
+                }
+            }
+        },
+        addElement(element) {
+            // Get the index of the corresponding element in the series list
+            const index = this.parentObject.configuration.series.findIndex(
+                series => series.identifier.key === element.identifier.key
+            );
+            let yAxisId = this.parentObject.configuration.series[index].yAxisId;
+            if (yAxisId === undefined) {
+                yAxisId = Y_AXIS_1;
+                this.composition.reorder(index, this.yAxis1.length);
+            }
+
+            const keyString = this.openmct.objects.makeKeyString(element.identifier);
+
+            // Store the element in the cache and set its yAxisId
+            this.elementsCache[keyString] =
+            JSON.parse(JSON.stringify(element));
+            if (this.elementsCache[keyString].yAxisId !== yAxisId) {
+                this.elementsCache[keyString].yAxisId = yAxisId;
+                // Mutate the YAxisId on the domainObject itself
+                this.mutateYAxisId(element, yAxisId);
```

Comment:
Only mutate the yAxisId if it doesn't already exist in the configuration. If it already exists, we should just assign it to the elementsCache and move on.

Reply:
I tried to make this change and it breaks the YAxis bucketing. I spent some time but still not understanding what's happening here. Maybe we can take another look together.